test/integration/operator/extension/extension/extension_suite_test.go

```diff
@@ -208,8 +214,63 @@ var _ = BeforeSuite(func() {
 		Expect(mgr.Start(mgrContext)).To(Succeed())
 	}()
 
+	go func() {
+		defer GinkgoRecover()
+		namespaceDeleter(mgrContext)
+	}()
+
 	DeferCleanup(func() {
 		By("Stop manager")
 		mgrCancel()
 	})
 })
+
+// namespaceDeleter finalises deletion of namespaces.
+// This is normally done by the kube-controller-manager which is not running in the integration test environment.
+func namespaceDeleter(ctx context.Context) {
```

Comment:
You can use https://github.com/gardener/gardener/blob/master/test/utils/namespacefinalizer/namespacefinalizer.go instead.